src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -45,30 +45,17 @@ public AddCommand parse(String args) throws ParseException {
                         PREFIX_AVAILABILITY, PREFIX_ANIMAL_TYPE, PREFIX_HOUSING);
 
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL)
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL,
+                PREFIX_ANIMAL_NAME, PREFIX_AVAILABILITY, PREFIX_ANIMAL_TYPE, PREFIX_HOUSING)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
-        if (argMultimap.getValue(PREFIX_AVAILABILITY).isPresent()
-                && argMultimap.getValue(PREFIX_AVAILABILITY).get().equals("nil")) {
-            if (argMultimap.getValue(PREFIX_ANIMAL_NAME).isPresent()
-                    && !argMultimap.getValue(PREFIX_ANIMAL_NAME).get().equals("nil")) {
-                throw new ParseException("Animal name should be 'nil' when availability is 'nil'.");
-            }
-
-            if (argMultimap.getValue(PREFIX_ANIMAL_TYPE).isPresent()
-                    && !argMultimap.getValue(PREFIX_ANIMAL_TYPE).get().equals("nil")) {
-                throw new ParseException("Animal type should be 'nil' when availability is 'nil'.");
-            }
-        }
-
-        if (argMultimap.getValue(PREFIX_ANIMAL_NAME).isPresent()
-                && !argMultimap.getValue(PREFIX_ANIMAL_NAME).get().equals("nil")) {
-            if (argMultimap.getValue(PREFIX_AVAILABILITY).isPresent()
-                    && argMultimap.getValue(PREFIX_AVAILABILITY).get().equals("Available")) {
-                throw new ParseException("Availability cannot be 'Available' when an animal name "
-                        + "is provided; animal name should be 'nil'.");
+        if (!argMultimap.getValue(PREFIX_ANIMAL_NAME).get().equals("nil")) {
+            if (argMultimap.getValue(PREFIX_AVAILABILITY).get().equals("Available")
+                    || argMultimap.getValue(PREFIX_AVAILABILITY).get().equals("nil")) {
```

Comment:
minor suggestion to improve readabillity
```java
if (!argMultimap.getValue(PREFIX_ANIMAL_NAME).get().equals("nil")) {
    String avail= argMultimap.getValue(PREFIX_AVAILABILITY).get();
    if (avail.equals("Available") || avail.equals("nil")) {
        throw ...
    }
}
```

Reply:
Changed